### PR TITLE
Mark external links with icon

### PR DESCRIPTION
**Pull request by dshevtsov**
Apply the same style to external links in navigation as was applied for external links in the content:
- page footer
- main navigation
- side navigation
- landing page

Also, applying this to all external links without inclusion. HTTP links are not allowed by the link checker, no need to exclude them.
Some links will be falsely positive when they are added as absolute links instead of internal. This is a violation of style and I don't think such a violation should be considered a valid requirement for this feature. A What's New page is the only common case that will contain many false positive external links without violation of a style. But this is only one page and we can take it in my opinion.
Otherwise, it's not easy to exclude only links for a particular website. I can exclude *docs.magento.com but this will introduce false negative cases such as a link to docs.magento.com from devdocs.magento.com will not be marked as external.

Previews:
- https://devdocs.magedevteam.com/741/
- https://docs.magedevteam.com/309/user-guide/
- https://docs.magedevteam.com/63/mbi/
- https://omsdocs.magedevteam.com/76/

### CODE REVIEW

**Comment by dshevtsov** [comment]
Got the icon in places where expected. There is one unexpected case of appearance that needs to be fixed - at the end of each topic.
Screen Shot 2022-09-30 at 3 42 19 PM

**Comment by bdenham** [comment]
@dshevtsov, is this PR ready for review or are you still working on things? It's currently still marked as Draft.

**Comment by dshevtsov** [comment]
Still working. And I could use help in identifying the bug from [my previous comment](https://github.com/magento-devdocs/devdocs-theme/pull/111#issuecomment-1264008072).

File: _includes/layout/navigation.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
                         <div class="nav-section">
                                 <div class="spectrum-Menu-sectionHeading">Cloud basics</div>
                                 <ul class="spectrum-Menu">
-                                        <li class="spectrum-Menu-item" role="menuitem"><a href="/cloud/bk-cloud.html"
+                                        <li class="spectrum-Menu-item" role="menuitem"><a href="https://stackoverflow.com/questions/2797091/css-and-and-or"
```
Comment (jeff-matthews) [inline _includes/layout/navigation.html:10]: This seems odd. Why are we linking to stackoverflow?

**Review by jeff-matthews — COMMENTED** [review]
The staging links look good. The only time I see links with external icons when I'm not expecting it is when the link points to a production site instead of the staging environment. I saw this in the user guide, but I suspect the link syntax in the .md file is using an absolute link instead of variables.

I only have one question about a change in the `_includes/layout/navigation.html` file.

File: _includes/layout/navigation.html (hunk 1 above)
Comment (dshevtsov) [reply _includes/layout/navigation.html:10]: No, I have added this for testing, because there were no external links there.

Comment (dshevtsov) [reply _includes/layout/navigation.html]: Reverted, sorry for confusion

Comment (jeff-matthews) [reply _includes/layout/navigation.html]: No problem! It make sense now. Thanks for reverting.

**Review by jeff-matthews — DISMISSED** [review]

**Comment by dshevtsov** [comment]
# Side navigation

Screen Shot 2022-10-04 at 9 44 20 AM

# Page footer

Screen Shot 2022-10-04 at 9 45 18 AM

# Main menu (top nav)

Screen Shot 2022-10-04 at 9 44 03 AM

**Comment by dshevtsov** [comment]
Do we need this feature in the app switcher?

**Comment by jeff-matthews** [comment]
> Do we need this feature in the app switcher?

I suppose it would provide consistency. If it can be done without a lot of work, then it's probably a good idea. If it'll take a lot of work, I think it's ok to skip it.

**Review by meker12 — COMMENTED** [review]
Looks good on the menu. 



For the top-level nav, the Development menu shows the ext-link icon, but they don't show up on the Setup, Testing, Functional Areas, and Tutorials.

**Comment by jeff-matthews** [comment]
The app switcher shows external links for the same links no matter what subdomain you're on. Is that expected? I would expect the icon placement to change depending on which subdomain you're on.

**Comment by dshevtsov** [comment]
@meker12, try now please

**Comment by dshevtsov** [comment]
> The app switcher shows external links for the same links no matter what subdomain you're on. Is that expected? I would expect the icon placement to change depending on which subdomain you're on.

Good point. I don't know an easy way to fix that. Reverting the change.

**Comment by dshevtsov** [comment]
Updated description and preview links

**Review by jeff-matthews — APPROVED** [review]

**Comment by dshevtsov** [comment]
@meker12, do you have any issues with your updates using the latest changes?

**Review by meker12 — APPROVED** [review]

**Pull request merged**